modules/EasyAbp.PaymentService.WeChatPay/src/EasyAbp.PaymentService.WeChatPay.Domain/EasyAbp/PaymentService/WeChatPay/PaymentServiceWeChatPayHandler.cs

```diff
@@ -32,17 +33,19 @@ public class PaymentServiceWeChatPayHandler : IWeChatPayHandler, ITransientDepen
             _paymentRepository = paymentRepository;
         }
         
-        public virtual async Task HandleAsync(XmlDocument xmlDocument)
+        public virtual async Task HandleAsync(WeChatPayHandlerContext context)
         {
-            using var disabledDataFilter = _dataFilter.Disable<IMultiTenant>();
+            var xmlDocument = context.WeChatRequestXmlData;
             
             var reader = new XmlNodeReader(xmlDocument.SelectSingleNode("xml") ?? throw new NullReferenceException());
 
-            if (reader["return_code"] != "SUCCESS" || reader["device_info"] != PaymentServiceWeChatPayConsts.DeviceInfo)
```

Comment:
Why removing `reader["return_code"] != "SUCCESS"`?

Reply:
Check return_code will cause payment.CancelPayment(_clock.Now); unreachable.
Maybe something like if (!context.IsSuccess || reader["device_info"] != PaymentServiceWeChatPayConsts.DeviceInfo) ?

Reply:
> Checking return_code will cause payment.CancelPayment(_clock.Now) unreachable.

Good, I think it should be considered and redesigned, but for now, maybe keeping it simple is a better way.

By the way, `context.IsSuccess` should not be check.

Reply:
> Good, I think it should be considered and redesigned, but for now, maybe keeping it simple is a better way.
So I should delete `reader["return_code"] != "SUCCESS"` or not?

> By the way, context.IsSuccess should not be check.
Out of curiosity, What is the meaning of `context.IsSuccess`? I thought it indicated result of payment call.

Reply:
You should see the commit: https://github.com/EasyAbp/Abp.WeChat/commit/5862be192b53e2cf11c6cc81f7d102d8be224a51